tests/baselines/reference/privateNamesInNestedClasses-1.errors.txt

```diff
@@ -0,0 +1,38 @@
+tests/cases/conformance/classes/members/privateNames/privateNamesInNestedClasses-1.ts(13,18): error TS2339: Property '#foo' does not exist on type 'A'.
+tests/cases/conformance/classes/members/privateNames/privateNamesInNestedClasses-1.ts(13,18): error TS18014: The property '#foo' cannot be accessed on type 'A' within this class because it is shadowed by another private identifier with the same spelling.
+
+
+==== tests/cases/conformance/classes/members/privateNames/privateNamesInNestedClasses-1.ts (2 errors) ====
+    // @target es6
+    
+    class A {
+       #foo = "A's #foo";
+       #bar = "A's #bar";
+       method () {
+           class B {
+               #foo = "B's #foo";
+               bar (a: any) {
+                   a.#foo; // OK, no compile-time error, don't know what `a` is
+               }
+               baz (a: A) {
+                   a.#foo; // compile-time error, shadowed
+                     ~~~~
+!!! error TS2339: Property '#foo' does not exist on type 'A'.

```

Comment:
this error still needs to be removed.

Reply:
Does the 'not found' message need to be removed? I agree the 'not found' is slightly misleading, but it *is* accurate in a way that teaches users about the feature. The three additional error messages we give below (with related spans) should make what's going on clear.

Shadowing is an edge case unlikely to be hit often, and we already throw a lot of code at it.
- Note that a 'not found'-style runtime error message is *all* Chrome provides in cases of shadowing, so we're already adding a ton of value:

![canary](https://user-images.githubusercontent.com/4691093/61366195-03d44000-a881-11e9-8bde-39e55c402e72.PNG)


Getting rid of the 'not found' might require some rearchitecture because:

`checkPropertyAccessExpressionOrQualifiedName`:
- calls `getPropertyForPrivateIdentifier`. The returned value is `undefined` in several cases. In all these cases, the property doesn't exist on the type.
    - In the shadowing case (which is very rare!) we still return 'undefined' but also provide the diagnostics below about shadowing
- later, `checkPropertyAccessExpressionOrQualifiedName` reports the missing property error when the property is undefined.
- `checkPropertyAccessExpressionOrQualifiedName` does not currently know anything about shadowing or do many early exits. So informing it of this edge case would involve doing at least one of these things (in my understanding):
    - enriching the return value of `getPropertyForPrivateIdentifier` with an object containing 'wasShadowed'. I actually like this option, but it's not idiomatic in this codebase. Very happy to do it if you think it's OK.
    - breaking up the logic for shadowing detection into a separate function and calling that function from `checkPropertyAccessExpressionOrQualifiedName`. This leads to some repeat work for looking up symbols and containing classes, and I know we're perf-oriented. It also increases the number of places that care about shadowing, which is really hard to explain. Everyone I show it to is like "Why would you nest classes?"

fwiw my vote would be, in order of preference:
- use a rich return value in getPropertyForPrivateIdentifier and get rid of the 'not found' message. If the cost of unidiomatic code is not too high.
- keep the 'not found'

Reply:
Sorry, I forgot about your earlier discussion [1]. Thanks for the detailed architectural walkthrough. You are right that shadowing is going to be a rare error, so let's just keep the duplicated error for now.

[1] At this point, my modus operandi is to open the PR page, and scroll back until I see comments that look old. Github's interface *really* doesn't scale to large PRs :(